src/saveload/oldloader_sl.cpp

```diff
@@ -1353,7 +1353,7 @@ bool LoadOldVehicle(LoadgameState *ls, int num)
 		if (_cargo_count != 0 && CargoPacket::CanAllocateItem()) {
 			StationID source =    (_cargo_source == 0xFF) ? INVALID_STATION : _cargo_source;
 			TileIndex source_xy = (source != INVALID_STATION) ? Station::Get(source)->xy : (TileIndex)0;
-			v->cargo.Append(new CargoPacket(_cargo_count, _cargo_periods, source, INVALID_STATION, source_xy));
+			v->cargo.Append(new CargoPacket(_cargo_count, _cargo_periods, source, source_xy, 0));
```

Comment:
0 is feeder share, I think, but we lose the first station? Why?

Reply:
confusingly, `source` is `first_station`. The `next_station` is removed, as it is always `INVALID_STATION`. It was really confusing code. Now it is simpler:

`count, period, first_station, source_xy, feeder_share`, for all of them. No optional parameters.